src/services/endpoint/EndpointService.ts

```diff
@@ -27,7 +28,7 @@ export class EndpointService {
               private socketBindingService: SocketBindingService, private launchType: LaunchTypeService) {
   }
 
-  getAllEndpoints(profileName: string, socketBindingGroup: string): ng.IPromise<IEndpoint[]> {
+  getAllEndpoints(profileName: string, socketBindingGroup: string, cacheContainer:ICacheContainer): ng.IPromise<IEndpoint[]> {
```

Comment:
There is no need to pass all three of these arguments as they are all contained within ICacheContainer.